src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java

```diff
@@ -0,0 +1,256 @@
+
+    /** The earliest new major version to which the database can be migrated. */
+    Semver earliestNewVersion; // e.g. 6.0.0 --> this is also the target version
+
+    /** The latest version before another major update is needed. */
+    Semver latestNewVersion; // e.g. 7.0.0
+
+    /** The error message to be displayed if the migration cannot proceed due to version incompatibility. */
+    String errorMessage;
+
+    /**
+     * Constructs a MigrationPath instance by defining the required version for the migration,
+     * and automatically determining the earliest and latest new versions based on major version
+     * increments.
+     *
+     * @param requiredVersion The minimum version required to start the migration, in string format.
+     *                            The earliestNewVersion is derived as the next major version after
+     *                            this required version, and latestNewVersion is the next major version
+     *                            after earliestNewVersion.
+     */
+    public MigrationPath(String requiredVersion) {
+        this.earliestNewVersion = new Semver(requiredVersion).nextMajor();
+        this.requiredVersion = new Semver(requiredVersion);
+        this.latestNewVersion = earliestNewVersion.nextMajor();
+        this.errorMessage = "Cannot start Artemis. Please start the release " + requiredVersion + " first, otherwise the migration will fail";
+    }
+}
+
+/**
+ * Handles the database migration process by checking against defined migration paths to ensure
+ * that the database schema is up-to-date before application startup. This class plays a critical
+ * role in optimizing application startup time by merging Liquibase migrations and eliminating Java
+ * migrations for each major version release.
+ * <p>
+ * This mechanism is essential for maintainers, aiming to streamline the startup, especially
+ * beneficial for server and e2e tests. It mandates administrators to upgrade to the latest minor
+ * version before a major update, facilitating a smooth transition and incorporation of all
+ * database changes.
+ * <p>
+ * Steps for new database migration paths:
+ * <ol>
+ * <li>Re-create the new initial scheme from the database after applying all existing liquibase migrations.</li>
+ * <li>Delete all existing liquibase migrations except the new initial scheme and *_cleanup.xml (adapt *_cleanup.xml with the latest date).</li>
+ * <li>Delete all Java migrations that have been executed before.</li>
+ * <li>Check that the new initial scheme is compatible with MySQL and Postgres and can be started from scratch in both environments.</li>
+ * <li>Add the new migration path in the constructor of DatabaseMigration.</li>
+ * <li>Verify that the migration works with the designated path (ideally with a local database and with a dump from a test / production system).</li>
+ * <li>Document the migration carefully in the release notes and the online documentation.</li>
```

Comment:
```suggestion
 * <li>Apply all existing liquibase migrations.</li>
 * <li>Re-create the new initial scheme from the database.</li>
 * <li>Delete all existing liquibase migrations except the new initial scheme and *_cleanup.xml</li>
 * <li>Rename *_cleanup.xml to match the latest date.</li>
 * <li>Delete all Java migrations that have been executed before.</li>
 * <li>Check that the new initial scheme is compatible with MySQL and Postgres and can be started from scratch in both environments.</li>
 * <li>Add the new migration path in the constructor of DatabaseMigration.</li>
 * <li>Verify that the migration works with the designated path (ideally with a local database and with a dump from a test / production system).</li>
 * <li>Document the migration carefully in the release notes and the online documentation.</li>
```
Questions that I recommend answering in that section:

1 How does on apply the existing liquibase migrations? Are you talking about starting a running application? Starting a new application? Executing a command?
2 How? => Gradle & the command
3 Where does the *_cleanup.xml come from? Auto-Generated? If it's not, why can the content stay as it is?
5 Why "that have been executed before"? After running Liquibase (unless via command) I would expect that all migrations have been run.
7 How do I do that? What version do I select? Looking at the versions above, it's the last version before the new release?
8 What does "verify" mean? No errors on start-up? Should one go through all of the schema? Or just make sure that basically nothing changed, including data?